lib/wibox/widget/checkbox.lua

```diff
@@ -0,0 +1,253 @@
+-- If none is set, then the `shape` property will be used.
+-- @beautiful beautiful.checkbox_check_shape
+-- @see gears.shape
+
+--- The padding between the outline and the progressbar.
+-- @beautiful beautiful.checkbox_paddings
+-- @tparam[opt=0] table|number paddings A number or a table
+-- @tparam[opt=0] number paddings.top
+-- @tparam[opt=0] number paddings.bottom
+-- @tparam[opt=0] number paddings.left
+-- @tparam[opt=0] number paddings.right
+
+--- The checkbox color.
+-- This will be used for the unchecked part border color and the checked part
+-- filling color. Note that `check_color` and `border_color` have priority
+-- over this property.
+-- @beautiful beautiful.checkbox_color
+
+--- The outer (unchecked area) border width.
+-- @property border_width
+
+--- The outer (unchecked area) background color, pattern or gradient.
+--@DOC_wibox_widget_checkbox_bg_EXAMPLE@
+-- @property bg
+
+--- The outer (unchecked area) border color.
+-- @property border_color
+
+--- The checked part border color.
+-- @property check_border_color
+
+--- The checked part border width.
+-- @property check_border_width
+
+--- The checked part filling color.
+-- @property check_color
+
+--- The outer (unchecked area) shape.
+--@DOC_wibox_widget_checkbox_shape_EXAMPLE@
+-- @property shape
+-- @see gears.shape
+
+--- The checked part shape.
+-- If none is set, then the `shape` property will be used.
+--@DOC_wibox_widget_checkbox_check_shape_EXAMPLE@
+-- @property check_shape
+-- @see gears.shape
+
+--- The padding between the outline and the progressbar.
+-- @property paddings
+-- @tparam[opt=0] table|number paddings A number or a table
+-- @tparam[opt=0] number paddings.top
+-- @tparam[opt=0] number paddings.bottom
+-- @tparam[opt=0] number paddings.left
+-- @tparam[opt=0] number paddings.right
+
+--- The checkbox color.
+-- This will be used for the unchecked part border color and the checked part
+-- filling color. Note that `check_color` and `border_color` have priority
+-- over this property.
+-- @property color
+
+local function outline_workarea(self, width, height)
+    local offset = (self._private.border_width or
+        beautiful.checkbox_border_width or 1)/2
+
+    return {
+        x      = offset,
+        y      = offset,
+        width  = width-2*offset,
+        height = height-2*offset
+    }
+end
+
+-- The child widget area
+local function content_workarea(self, width, height)
+    local padding = self._private.paddings or {}
+    local offset = self:get_check_border_width() or 0
+    local wa = outline_workarea(self, width, height)
+
+    wa.x      = offset + wa.x + (padding.left or 1)
+    wa.y      = offset + wa.y + (padding.top  or 1)
+    wa.width  = wa.width  - (padding.left or 1) - (padding.right  or 1) - 2*offset
+    wa.height = wa.height - (padding.top  or 1) - (padding.bottom or 1) - 2*offset
+
+    return wa
+end
+
+local function draw(self, context, cr, width, height)
+    local size = math.min(width, height)
```

Comment:
Could `draw` "really use" the provided width/height instead of using the minimum of them? If some layout ignores the fit result and gives this widget a non-rectangular area, some part would otherwise be left "empty" and this could be confusing to users.